clang-tools-extra/clang-tidy/llvmlibc/CalleeNamespaceCheck.cpp

```diff
@@ -45,18 +46,21 @@ void CalleeNamespaceCheck::check(const MatchFinder::MatchResult &Result) {
   if (FuncDecl->getBuiltinID() != 0)
     return;
 
-  // If the outermost namespace of the function is __llvm_libc, we're good.
+  // If the outermost namespace of the function is a macro that starts with
+  // __llvm_libc, we're good.
   const auto *NS = dyn_cast<NamespaceDecl>(getOutermostNamespace(FuncDecl));
-  if (NS && NS->getName() == "__llvm_libc")
+  if (NS && Result.SourceManager->isMacroBodyExpansion(NS->getLocation()) &&
```

Comment:
To be honest I was thinking more about something that would utilize isExpandedFromMacro.
Most probably this could be done by few custom AST matchers.

Reply:
I'm not entirely sure what you mean here. I tried adding `unless(isExpandedFromMacro(RequiredNamespaceMacroName.str()))` to the `declRefExpr` above, but it caused the matcher to fail the `wrong_name_macro_func` check.

Reply:
I can improve this in a followup patch if necessary.